ladybug/src/main/java/nl/nn/ibistesttool/IbisDebuggerAdvice.java

```diff
@@ -189,58 +177,66 @@ public PipeRunResult debugPipeGetInputFrom(ProceedingJoinPoint proceedingJoinPoi
-		}
-		if (result!=null && result instanceof PipeRunResult) {
-			PipeRunResult prr = (PipeRunResult)result;
-			prr.setResult(ibisDebugger.senderOutput(sender, messageId, prr.getResult()));
-			return result;
+			result = ibisDebugger.preserveInput(messageId, (Message)result);
 		}
-		return (M)ibisDebugger.senderOutput(sender, messageId, Message.asMessage(result));
+		return result;
 	}
 	
 	/**
 	 * Provides advice for {@link ISender#sendMessage(Message message, IPipeLineSession session)}
 	 */
 	public Message debugSenderInputOutputAbort(ProceedingJoinPoint proceedingJoinPoint, Message message, IPipeLineSession session) throws Throwable {
-		return (Message)debugSenderInputOutputAbort(proceedingJoinPoint, message, session, 0);
+		Object result = debugSenderInputStubAbort(proceedingJoinPoint, message, session, 0);
+		ISender sender = (ISender)proceedingJoinPoint.getTarget();
+		String messageId = session == null ? null : session.getMessageId();
+		return ibisDebugger.senderOutput(sender, messageId, Message.asMessage(result));
 	}
 
 	/**
 	 * Provides advice for {@link IBlockEnabledSender#sendMessage(Object blockHandle, Message message, IPipeLineSession session)}
 	 */
 	public Message debugBlockEnabledSenderInputOutputAbort(ProceedingJoinPoint proceedingJoinPoint, Object blockHandle, Message message, IPipeLineSession session) throws Throwable {
-		return (Message)debugSenderInputOutputAbort(proceedingJoinPoint, message, session, 1);
+		Object result = debugSenderInputStubAbort(proceedingJoinPoint, message, session, 1);
+		ISender sender = (ISender)proceedingJoinPoint.getTarget();
+		String messageId = session == null ? null : session.getMessageId();
+		return ibisDebugger.senderOutput(sender, messageId, Message.asMessage(result));
 	}
 
 	/**
 	 * Provides advice for {@link IStreamingSender#sendMessage(Message message, IPipeLineSession session, IForwardTarget next)}
 	 */
 	public PipeRunResult debugStreamingSenderInputOutputAbort(ProceedingJoinPoint proceedingJoinPoint, Message message, IPipeLineSession session, IForwardTarget next) throws Throwable {
-		return (PipeRunResult)debugSenderInputOutputAbort(proceedingJoinPoint, message, session, 0);
+		Object result = debugSenderInputStubAbort(proceedingJoinPoint, message, session, 0);
+		PipeRunResult prr;
+		if (result instanceof PipeRunResult) {
+			prr = (PipeRunResult)result;
+			result = prr.getResult();
+		} else {
+			// Create PipeRunResult when streaming sender is stubbed, this will forward to the next pipe and process the
+			// message in a streaming.auto=false way (also when at the time of the original report the message was
+			// processed with streaming.auto=true)
+			prr = new PipeRunResult();
+		}
+		ISender sender = (ISender)proceedingJoinPoint.getTarget();
+		String messageId = session == null ? null : session.getMessageId();
+		prr.setResult(ibisDebugger.senderOutput(sender, messageId, Message.asMessage(result)));
+		return prr;
```

Comment:
Dit riep debugSenderInputOutputAbort(). Is dat niet meer nodig?

Reply:
Van de debugSenderInputOutputAbort() heb ik het Output stuk logica verwijderd en daarom de methode hernoemd naar debugSenderInputStubAbort() die nog steeds aangeroepen wordt. Het Output stuk zit nu in de verschillende methodes zelf omdat de ene methode met Message werkt en de ander met PipeRunResult.

Reply:
zie comment thread hierboven